### ISSUE TITLE
indexing failed with 504 Gateway Timeout

### ISSUE THREAD

**Issue by wmalik** [miku/esbulk#22]
esbulk fails with the following error while indexing a file with around 850k JSON documents:

```
08:19:47.337 2018/07/06 07:01:38 indexing failed with 504 Gateway Timeout:
```

In this case, it could make sense to retry the HTTP request with a backoff until elasticsearch returns a 200 response. I noticed that currently the HTTP requests are not retried: https://github.com/miku/esbulk/blob/master/indexing.go#L174

Would it make sense to retry the HTTP requests N times with an exponential back-off?

**Comment by wmalik** [comment]
@miku thank you very much for looking into it, and thanks for esbulk in general :)

Please do let me know if I can help with the implementation/testing etc.

**Comment by miku** [comment]
@wmalik I switched the http client to the default [Client](https://github.com/sethgrid/pester/blob/03e26c9abbbf5accb8349790bf9f41bde09d72c3/pester.go#L122) from [pester](https://github.com/sethgrid/pester), which offers 3 retries with a linear backoff. Not optimal, but a first step.

This change is included in [0.5.1](https://github.com/miku/esbulk/releases/tag/v0.5.1).

**Comment by miku** [comment]
Closing this, since I assume this is less of a problem with the new `pester` client.
